lib/linguist/vocabulary.ex

```diff
@@ -79,6 +79,8 @@ defmodule Linguist.Vocabulary do
             source
         end
 
+      name = name |> to_string() |> String.to_atom()
```

Comment:
I'm confused about this change since in your description you said 

> locales() return list of binaries instead of atoms (because existing tests say that, but honestly I prefer atoms here).

Reply:
`Compiler.compile` requires @locales to be a KeyWord list, so you cannot pass here locales as strings, they should be atoms. That's why I convert locale name into atom here.
`locales/0` function is being generated by compiler and it always return list of string, no matter how locales were passed into Compiler.

Although, it is possible not to check if locales are Keyword list, instead write a check if list elements are `{name, paths}` tuple. This does no require to convert locale names into atoms.

Reply:
Previous solution was a bit tricky, so I've made few changes.